### ISSUE TITLE
transfer should not be required to be set when in and output primaries match

### ISSUE THREAD

**Issue by Lypheo** [sekrit-twc/zimg#126]
Currently, zimg considers a colorspace with unspecified transfer characteristics to be invalid when primaries are set. This makes zimg reject conversions from YCbCr to RGB where input primaries are known but transfer is not, which doesn’t make sense because the transfer curve is irrelevant for a (non-constant luminance) YUV->RGB operation.
The simplest way to reproduce is in VS with:
```py
out = core.resize.Bicubic(core.std.BlankClip(format=vs.YUV444PS), format=vs.RGB48, matrix_in=1, primaries_in=1, primaries=1)
```
This matters, at least in VS, because source filters attach color properties to the clip based on the tags in the file, so this for example would fail, much to the confusion of an unsuspecting user, if the file has a primaries tag but no transfer one, even though it works if it has neither:
```py
src = core.ffms2.Source("…")
out = core.resize.Bicubic(src, format=vs.RGB48, matrix_in_s="709")
out.set_output()
```
https://github.com/Lypheo/zimg/commit/94c6a864f62bc73639a55d751731d17c43a24386 seems to fix it, but I may have missed something.

**Comment by sekrit-twc** [comment]
We can't do this in general, because just like 2020 CL vs NCL, there could hypothetically be a transfer function that depends on the gamut.

**Comment by Lypheo** [comment]
I don’t think I understand. The transfer curve isn’t even changed in the example above. The issue lies in how zimg sets requirements for the conversion by checking the in/out colorspaces independently instead of the actual graph, resulting in overly strict requirements for conversions where in and out parameters match and, thus, don’t matter at all.

**Comment by sekrit-twc** [comment]
Upon further review, it looks like the key safeguards are already in the graph logic, which prevents anything from being converted to/from UNSPEC. Implemented in 8cdea30
